sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -520,6 +520,22 @@ class FileIndexSuite extends SharedSparkSession {
       SQLConf.get.setConf(StaticSQLConf.METADATA_CACHE_TTL_SECONDS, previousValue)
     }
   }
+
+  test("SPARK-28098 - supporting read partitioned Hive tables with subdirectories") {
```

Comment:
not sure what this is testing for

this case also passes on 2.4 without any patched code